File: python-srm/srmtest.c
#include "libsrm_v1_1.h"

/* Standalone C client for testing */

int main() {
  char **surls;
  char **prots;
  int reqid;
  struct srm_filestatus *fs;
  struct srm_filemetadata *fm;
  int j, i;

  surls = (char**)malloc(sizeof(char*)*2);
  surls[0] = (char*)malloc(sizeof(char)*200);
  surls[1] = (char*)malloc(sizeof(char)*200);
  prots = (char**)malloc(sizeof(char*));
  prots[0] = (char*)malloc(sizeof(char)*200);

  sprintf(surls[0], "srm://castorsrm.cern.ch:8443/castor/cern.ch/grid/atlas/f1");
  sprintf(surls[1], "srm://castorsrm.cern.ch:8443/castor/cern.ch/grid/atlas/f2");
  sprintf(prots[0], "gsiftp");

  printf("ping result %d\n", srm_ping("srm://castorsrm.cern.ch:8443"));

  j = srm_get("srm://castorsrm.cern.ch:8443", 2, surls, 1, prots, &reqid, &fs);
  printf("get count is %d\n", j);
  for (i=0; i<j; i++) {
    printf("SURL is %s\n", fs[i].surl);
    printf("STATUS is %d\n", fs[i].status);
    if (fs[i].status != 0)
      printf("TURL is %s\n", fs[i].turl);
  }

  j = srm_getfilemetadata("srm://castorsrm.cern.ch:8443", 2, surls, &fm);
  printf("getfilemetadata count is %d\n", j);
  for (i=0; i<j; i++) {
    printf("SURL is %s\n", fm[i].surl);
    printf("isPinned is %d\n", fm[i].isPinned);
    printf("size is %ld\n", fm[i].size);
  }
}

File: python-srm/srmtest.py
# Standalone Python client for testing

import sys
import SRM

#print "TESTING SINGLE FILE"
#print SRM.srmGet('srm://castorsrm.cern.ch:8443/', ['srm://castorsrm.cern.ch:8443/castor/cern.ch/grid/atlas/f1'], ['gsiftp'])
#
#print "TESTING BULK SRM GET"
#(reqid, fs) = SRM.srmGet('srm://castorsrm.cern.ch:8443/', ['srm://castorsrm.cern.ch:8443/castor/cern.ch/grid/atlas/f1', 'srm://castorsrm.cern.ch:8443/castor/cern.ch/grid/atlas/f1'], ['gsiftp'])
#print reqid, fs
#
#print "TESTING SRM GET REQUEST STATUS"
#print SRM.srmGetRequestStatus('srm://castorsrm.cern.ch:8443/', reqid)
#
print "TESTING SRM GET FILE METADATA"
print SRM.srmGetFileMetaData('srm://castorsrm.cern.ch:8443', ['srm://castorsrm.cern.ch:8443/castor/cern.ch/grid/atlas/rome.003014.simul.M1_minbias._00001.pool.root'])
